linera-views/src/common.rs

```diff
@@ -123,6 +150,61 @@ pub trait KeyValueStoreClient {
     /// Retrieves a `Vec<u8>` from the database using the provided `key`.
     async fn read_key_bytes(&self, key: &[u8]) -> Result<Option<Vec<u8>>, Self::Error>;
 
+    /// Retrieves a `Vec<u8>` from the database using the provided `key`.
+    async fn read_key_big_bytes(&self, key: &[u8]) -> Result<Option<Vec<u8>>, Self::Error>
+    where
+        Self::Error: From<bcs::Error>,
+        Self::Error: From<DatabaseConsistencyError>,
+    {
+        let mut big_key = vec![0];
+        big_key.extend(key);
+        let value = self.read_key_bytes(&big_key).await?;
+        if value.is_none() {
+            return Ok(None);
+        }
+        let value = value.unwrap();
+        if value.is_empty() {
+            return Err(DatabaseConsistencyError::ZeroLengthValue.into());
+        }
+        let first_value = *value.first().unwrap();
+        if first_value != 0 && first_value != 1 {
+            return Err(DatabaseConsistencyError::FirstValueDifferentZeroOne.into());
+        }
+        if first_value == 0 {
+            return Ok(Some(value[1..].to_vec()));
+        }
+        let count = bcs::from_bytes::<usize>(&value[1..])?;
+        let mut big_keys = Vec::new();
+        for i in 0..count {
+            let mut big_key = vec![1];
+            big_key.extend(key);
+            big_key.extend(bcs::to_bytes(&i)?);
+            big_keys.push(big_key);
+        }
+        let segments = self.read_multi_key_bytes(big_keys).await?;
+        let mut big_value = Vec::new();
+        for segment in segments {
+            match segment {
+                None => {
+                    return Err(DatabaseConsistencyError::MissingSegment.into());
+                }
+                Some(segment) => {
+                    big_value.extend(segment);
+                }
+            }
+        }
+        Ok(Some(big_value))
+    }
+
+    /// Reads a single `key` and deserialize the result if present.
+    async fn read_key_big<V: DeserializeOwned>(&self, key: &[u8]) -> Result<Option<V>, Self::Error>
+    where
+        Self::Error: From<bcs::Error>,
+        Self::Error: From<DatabaseConsistencyError>,
```

Comment:
```suggestion
        Self::Error: From<bcs::Error> + From<DatabaseConsistencyError>,
```

(Not important, but it's more consistent with the style elsewhere, I think.)